provider/pihole/clientV6.go

```diff
@@ -0,0 +1,376 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pihole
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/linki/instrumented_http"
+	log "github.com/sirupsen/logrus"
+
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/provider"
+)
+
+// piholeClient implements the piholeAPI.
+type piholeClientV6 struct {
+	cfg        PiholeConfig
+	httpClient *http.Client
+	token      string
+}
+
+// newPiholeClient creates a new Pihole API V6 client.
+func newPiholeClientV6(cfg PiholeConfig) (piholeAPI, error) {
+	if cfg.Server == "" {
+		return nil, ErrNoPiholeServer
+	}
+
+	// Setup an HTTP client
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: cfg.TLSInsecureSkipVerify,
+			},
+		},
+	}
+	cl := instrumented_http.NewClient(httpClient, &instrumented_http.Callbacks{})
+
+	p := &piholeClientV6{
+		cfg:        cfg,
+		httpClient: cl,
+	}
+
+	if cfg.Password != "" {
+		if err := p.retrieveNewToken(context.Background()); err != nil {
+			return nil, err
+		}
+	}
+
+	return p, nil
+}
+
+func (p *piholeClientV6) getConfigValue(ctx context.Context, rtype string) ([]string, error) {
+	apiUrl, err := p.urlForRecordType(rtype)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("Listing %s records from %s", rtype, apiUrl)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	jRes, err := p.do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse JSON response
+	var apiResponse ApiRecordsResponse
+	err = json.Unmarshal(jRes, &apiResponse)
+	if err != nil {
+		log.Errorf("error reading response: %s", err)
+	}
```

Comment:
if err := json.Unmarshal(jRes, &apiError); err != nil {
return nil, fmt.Errorf("failed to unmarshal error response: %w", err)
}